modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/BlackFxSingleBarrierOptionProductPricer.java

```diff
@@ -0,0 +1,450 @@
+
+    validate(option, ratesProvider, volatilityProvider);
+    SimpleConstantContinuousBarrier barrier = (SimpleConstantContinuousBarrier) option.getBarrier();
+    ResolvedFxVanillaOption underlyingOption = option.getUnderlyingOption();
+    if (volatilityProvider.relativeTime(underlyingOption.getExpiry()) < 0d) {
+      return 0d;
+    }
+    ResolvedFxSingle underlyingFx = underlyingOption.getUnderlying();
+    Currency ccyBase = underlyingFx.getBaseCurrencyPayment().getCurrency();
+    Currency ccyCounter = underlyingFx.getCounterCurrencyPayment().getCurrency();
+    CurrencyPair currencyPair = underlyingFx.getCurrencyPair();
+    DiscountFactors baseDiscountFactors = ratesProvider.discountFactors(ccyBase);
+    DiscountFactors counterDiscountFactors = ratesProvider.discountFactors(ccyCounter);
+
+    double rateBase = baseDiscountFactors.zeroRate(underlyingFx.getPaymentDate());
+    double rateCounter = counterDiscountFactors.zeroRate(underlyingFx.getPaymentDate());
+    double costOfCarry = rateCounter - rateBase;
+    double dfBase = baseDiscountFactors.discountFactor(underlyingFx.getPaymentDate());
+    double dfCounter = counterDiscountFactors.discountFactor(underlyingFx.getPaymentDate());
+    double spot = ratesProvider.fxRate(currencyPair);
+    double strike = underlyingOption.getStrike();
+    double forward = spot * dfBase / dfCounter;
+    double volatility = volatilityProvider.getVolatility(currencyPair, underlyingOption.getExpiry(), strike, forward);
+    double timeToExpiry = volatilityProvider.relativeTime(underlyingOption.getExpiry());
+    double price = BARRIER_PRICER.price(
+        spot, strike, timeToExpiry, costOfCarry, rateCounter, volatility, underlyingOption.getPutCall().isCall(), barrier);
+    if (option.getRebate().isPresent()) {
+      CurrencyAmount rebate = option.getRebate().get();
+      double priceRebate = rebate.getCurrency().equals(ccyCounter) ?
+          CASH_REBATE_PRICER.price(spot, timeToExpiry, costOfCarry, rateCounter, volatility, barrier.inverseKnockType()) :
+          ASSET_REBATE_PRICER.price(spot, timeToExpiry, costOfCarry, rateCounter, volatility, barrier.inverseKnockType());
+      price += priceRebate * rebate.getAmount() / Math.abs(underlyingFx.getBaseCurrencyPayment().getAmount());
+    }
+    return price;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value sensitivity of the FX barrier option product.
+   * <p>
+   * The present value sensitivity of the product is the sensitivity of {@link #presentValue} to
+   * the underlying curves.
+   * <p>
+   * The volatility is fixed in this sensitivity computation.
+   * 
+   * @param option  the option product
+   * @param ratesProvider  the rates provider
+   * @param volatilityProvider  the Black volatility provider
+   * @return the present value curve sensitivity of the product
+   */
+  public PointSensitivityBuilder presentValueSensitivity(
```

Comment:
We should probably rename the method to "presentValueSensitivityStickyStrike" (same in "BlackFxVanillaOptionProductPricer") to be coherent with "VolatilitySwaptionPhysicalProductPricer" and to allow the introduction of the 'SensitivityStickyMoneyness' and 'SensitivityStickyDelta' later. 
Add also in the Java doc the fact that this is 'StickyStrike' sensitivity.